capstone/src/test/java/com/google/step/clients/MetricsClientTests.java

```diff
@@ -0,0 +1,353 @@
+        entity2.setProperty("restaurantKey", "1");
+        entity2.setProperty("restaurantName", "Marlows");
+        entity2.setProperty("year", 2015);
+        entity2.setProperty("week", 7);
+        entity2.setProperty("count", 200);
+
+        datastoreService.put(entity1);
+        datastoreService.put(entity2);
+
+        WeeklyPageView resultPageView = metricsClient.getCurrentPageViews("Marlows", "1");
+        Assert.assertEquals("Year", year, resultPageView.getYear());
+        Assert.assertEquals("Week", week, resultPageView.getWeek());
+        Assert.assertEquals("Count", 100, resultPageView.getCount());
+    }
+
+    @Test
+    public void testGetCurrentPageViewsWithoutExistingEntity() {
+        // Tests if the method correctly created an entity and puts into dataStore
+        // if no entity exists for this week currently.
+        // Result: WeeklyPageView with all variables set to 0.
+
+        // Using calendar for this test is necessary since the method in the
+        // metricsClient uses the calendar as well to get the current date.
+        WeeklyPageView resultPageView = metricsClient.getCurrentPageViews("Marlows", "1");
+        Assert.assertEquals("Week", 0, resultPageView.getWeek());
+        Assert.assertEquals("Year", 0, resultPageView.getYear());
+        Assert.assertEquals("Count", 0, resultPageView.getCount());
+    }
+
+    @Test
+    public void testGetAllPageViews() {
+        // Tests the getAllPageViews method by adding several restaurant page view entities.
+        // Result: [
+        //              RestaurantPageView(
+        //                  name: "Subway"
+        //                  pageViews: [
+        //                      WeeklyPageView(
+        //                          week: 11
+        //                          year: 2015
+        //                          count: 1000
+        //                      )
+        //                 ]
+        //              ),
+        //              RestaurantPageView(
+        //                  name: "Marlows"
+        //                  pageViews: [
+        //                      WeeklyPageView(
+        //                          week: 13
+        //                          year: 2014
+        //                          count: 200
+        //                      ),
+        //                      WeeklyPageView(
+        //                          week: 13
+        //                          year: 2015
+        //                          count: 100
+        //                      ),
+        //                      WeeklyPageView(
+        //                          week: 15
+        //                          year: 2015
+        //                          count: 50
+        //                      ),
+        //                      WeeklyPageView(
+        //                          week: 20
+        //                          year: 2016
+        //                          count: 300
+        //                      )
```

Comment:
Related to my other comment about the sort order:
Where would the following entry fall in the order? Consider adding an entry like this to ensure the sort is working as expected.
WeeklyPageView(
    week: 8
    year: 2017
    count: 5
)

Reply:
Added! It worked, but I had to change the order of the sorts to sort first by year then by week. Great catch!